lib/problems.rb

```diff
@@ -1,13 +1,120 @@
 require_relative './stack.rb'
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def balanced(string)
```

Comment:
Since you're building `expected_mirror` up, this does have some space complexity.  It should be O(n) space complexity.